demos/tests/args.py

```diff
@@ -73,6 +73,27 @@ def resolve(self, context):
         return str(seq_dir / name_format)
 
 
+class DataOrigFileNamesArg:
+    def __init__(self, sequence_name):
+        self.sequence_name = sequence_name
+
+    def resolve(self, context):
+        seq_dir = context.data_sequence_dir / self.sequence_name
+        seq = [Path(data.resolve(context))
+            for data in context.data_sequences[self.sequence_name]]
+            
+        assert len(set(data.suffix for data in seq)) == 1, "all images in the sequence must have the same extension"
+        assert '%' not in seq[0].suffix
+
+        if not seq_dir.is_dir():
+            seq_dir.mkdir(parents=True)
+
+            for index, data in enumerate(context.data_sequences[self.sequence_name]):
+                shutil.copyfile(data.resolve(context), str(seq_dir / (seq[index].stem + seq[0].suffix)))
```

Comment:
Why re-resolve `data` when you already have the `seq` list where it's already resolved?